src/proxy/Map.ts

```diff
@@ -56,4 +124,269 @@ export class Map<K, V> extends BaseProxy implements IMap<K, V> {
     isEmpty(): Q.Promise<boolean> {
         return this.encodeInvokeOnRandomTarget<boolean>(MapIsEmptyCodec);
     }
+
+    getAll(keys: K[]): Q.Promise<any[]> {
+        var partitionService = this.client.getPartitionService();
+        var partitionsToKeys: {[id: string]: any} = {};
+        var key: K;
+        for (var i in keys) {
+            key = keys[i];
```

Comment:
Putting it inside for method does not make its scope restricted to for loop. So I intentionally put them outside. This way it is less likely that someone mistakenly introduces the same variable later in the same method expecting it to be null. Matter of taste.